### PR TITLE
add: unit tests

### PR DESCRIPTION
**Pull request by vidhi-mody**

### CODE REVIEW

**Comment by Serena-Wang** [comment]
All look good! Thanks, Vidhi!

Could you add a test case that a file path is not categorized to be any of those 4 PullRequestFileType?

**Review by Serena-Wang — CHANGES_REQUESTED** [review]
All look good! Thanks, Vidhi!

Could you add a test case that a file path is not categorized to be any of those 4 PullRequestFileType?

**Comment by vidhi-mody** [comment]
> All look good! Thanks, Vidhi!
> 
> Could you add a test case that a file path is not categorized to be any of those 4 PullRequestFileType?

The code categories a PR as `OTHER_NONFS` if it doesn't belong to any of the other three PullRequestFileTypes. So a PR will always fall into one of the 4 PullRequestFileTypes.

File: tests/test_metadata_content.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        #print(is_invalid, errors)
+        self.assertTrue(is_invalid)
+        self.assertIn("METADATA_ERROR: Model abreviation in metadata inconsistent with folder name", errors[0])
+
+    #Metadata is not a valid yaml file
+    def test_invalid_metadata_yaml_file(self):
+        is_invalid, errors = check_metadata_file(Path("tests/testfiles/data-processed/teamA-modelA/metadata-teamA-modelC.txt"))
+        #print(is_invalid, errors)
+        self.assertTrue(is_invalid)
+        self.assertIn("METADATA ERROR: Metadata YAML Fromat Error", errors[0])
```
Comment (Serena-Wang) [inline tests/test_metadata_content.py:30]: small typo in error msg here and a couple of places below

File: tests/test_metadata_content.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    #A metadata file is submitted with special characters and extra space in the end of the file
+    def test_invalid_metadata_file_format(self):
```
Comment (Serena-Wang) [inline tests/test_metadata_content.py:33]: I think special characters and extra spaces are not checked in the current validation.

 If you add this feature to metadata.py , please also commit that file. And maybe we should have a more informative error msg for this case.

**Review by Serena-Wang — CHANGES_REQUESTED** [review]
Thanks!

**Pull request merged**